node.js

```diff
@@ -49,7 +49,42 @@ module.exports = {
     'space-before-function-paren': off,
     'keyword-spacing': error,
     'eol-last': [ error, always ],
-    'brace-style': [ error, '1tbs', { allowSingleLine: true } ],
-    'curly': [ error, all ]
+    'brace-style': [ error, '1tbs', { allowSingleLine: false } ],
+    'curly': [ error, all ],
+    'block-spacing': [ error,  always ],
+    'indent': [ error, 2, { SwitchCase: 1 } ],
+    'no-tabs': error,
+    'spaced-comment': [ error, always ],
+    'array-callback-return': warn,
+    'eqeqeq': [ error, always, { null: 'ignore' } ],
+    'no-else-return': [ warn, { allowElseIf: false } ],
+    'yoda': error,
+    'no-throw-literal': error,
+    'no-self-compare': error,
+    'no-return-assign': error,
+    'no-loop-func': error,
+    'valid-jsdoc': warn,
+    'for-direction': error,
```

Comment:
Stop for loops for being unintentionally infinite (usually by using a counter that goes in the wrong direction)
https://eslint.org/docs/rules/for-direction
```js
// Old
for (let i = 0; i < 10; i--) {
}
// New
for (let i = 0; i < 10; i++) {
}
``` 
*Not auto-fixable but very very bad, so there shouldn't be any usage of this

Reply:
I didn't know such a check can be done with a linter 👍

Reply:
What if we want infinite loops?

This is a very common pattern in `redux saga`

```
function* watchRequests() {
  // 1- Create a channel for request actions
  const requestChan = yield actionChannel('REQUEST')
  while (true) {
    // 2- take from the channel
    const {payload} = yield take(requestChan)
    // 3- Note that we're using a blocking call
    yield call(handleRequest, payload)
  }
}
```

Would this be affected by the linting rule?

Also, Turing Award for solving the halting problem?

Reply:
This rule doesn't look at while loops, but https://github.com/appliedblockchain/eslint-config/pull/13#issuecomment-383886364 will mean you can still use `while (true)`